### DISCUSSION TITLE
How to set Modern.js Builder mode？

### DISCUSSION THREAD

**Discussion by Lzzzs** [web-infra-dev/modern.js#4936] [Q&A] [answered]
我在官方文档上没有看到，如何设置当前是啥环境，请问下怎么才可以设置

**Comment by chenjiahan** [comment]
目前 mode 是固定的，运行 modern dev 是开发环境，运行 modern build 是生产环境。

另外，你也可以通过 output.disableMinimize 来在 build 时关闭压缩，方便调试。

**Reply by Lzzzs** [reply]
但是我通过 process.env.NODE_ENV 去判断是否生产，然后在生产把 sourcemap 禁用，通过 build 打包之后并没有生效诶

![image](https://github.com/web-infra-dev/modern.js/assets/74575471/bf707657-8be1-499f-ba7a-4b30c95f163c)
![image](https://github.com/web-infra-dev/modern.js/assets/74575471/a82148e6-afc5-4588-945d-f8226cac30ef)

**Reply by chenjiahan** [reply, marked as answer]
我看了下是 Builder CLI 的 bug，没有在一开始就设置 NODE_ENV，所以你的 config 文件里没有拿到正确的 NODE_ENV 值。

你可以先手动处理一下哈，执行 `NODE_ENV=production builder build` 应该就可以。

我们会在 Rsbuild 里尽快修复这个问题，Rsbuild 是 Modern.js Builder 的新版本，参考 https://github.com/web-infra-dev/rsbuild

**Reply by chenjiahan** [reply]
修复 PR：https://github.com/web-infra-dev/rsbuild/pull/425

**Comment by Lzzzs** [comment]
也就是更推荐使用 Rsbuild吗。我是想迁移Rspack 的，看的官方的迁移指南，说 CRA 项目更适合通过 Modern.js Builder 去做迁移工作，才选择的 Modern.js Builder

**Reply by chenjiahan** [reply]
Rspack 文档里的迁移指南还未更新，我们会尽快更新哈

**Reply by chenjiahan** [reply]
推荐用 Rsbuild

**Reply by Lzzzs** [reply]
还有两个 Modern.js Builder 的小问题～

- 配置了 https：true。启动开发服务器时，network 不会生成对应 https 链接，只是 http
![image](https://github.com/web-infra-dev/modern.js/assets/74575471/9e381ec7-db29-4b5c-99d9-0bf4e2034fa9)
- 配置了startUrl：true。并不会自动打开浏览器

**Reply by chenjiahan** [reply]
- 第一个问题缺失存在，我们尽快修复下
- 第二个问题在 Rsbuild 里没复现，应该是已经修复了哈。

后面有新的问题可以直接到 Rsbuild 仓库提 issue ❤️
